presto-thrift-connector-api/src/main/java/com/facebook/presto/connector/thrift/api/datatypes/PrestoThriftBigint.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.api.datatypes;
+
+import com.facebook.presto.connector.thrift.api.PrestoThriftBlock;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.LongArrayBlock;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.facebook.presto.connector.thrift.api.PrestoThriftBlock.bigintData;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Elements of {@code nulls} array determine if a value for a corresponding row is null.
+ * Elements of {@code longs} array are values for each row. If row is null then value is ignored.
+ */
+@ThriftStruct
+public final class PrestoThriftBigint
+        implements PrestoThriftColumnData
+{
+    private final boolean[] nulls;
+    private final long[] longs;
+
+    @ThriftConstructor
+    public PrestoThriftBigint(@Nullable boolean[] nulls, @Nullable long[] longs)
+    {
+        checkArgument(sameSizeIfPresent(nulls, longs), "nulls and values must be of the same size");
+        this.nulls = nulls;
+        this.longs = longs;
+    }
+
+    @Nullable
+    @ThriftField(value = 1, requiredness = OPTIONAL)
+    public boolean[] getNulls()
+    {
+        return nulls;
+    }
+
+    @Nullable
+    @ThriftField(value = 2, requiredness = OPTIONAL)
+    public long[] getLongs()
+    {
+        return longs;
+    }
+
+    @Override
+    public Block toBlock(Type desiredType)
+    {
+        checkArgument(BIGINT.equals(desiredType), "type doesn't match: %s", desiredType);
+        int numberOfRecords = numberOfRecords();
+        return new LongArrayBlock(
+                numberOfRecords,
+                nulls == null ? new boolean[numberOfRecords] : nulls,
+                longs == null ? new long[numberOfRecords] : longs);
+    }
+
+    @Override
+    public int numberOfRecords()
+    {
+        return nulls != null ? nulls.length : (longs != null ? longs.length : 0);
```

Comment:
This line is really too complex for inline if.  Just use something simple like this:
```java
if (nulls != null) {
    return nulls.length;
}
if (longs != null) {
    return longs.length;
}
return 0;